cmd/snap/cmd_set_test.go

```diff
@@ -127,21 +127,51 @@ func (s *snapSetSuite) TestSnapSetFailParsingWithStrictJSON(c *check.C) {
 	c.Assert(err, check.ErrorMatches, "failed to parse JSON:.*")
 }
 
-func (s *snapSetSuite) TestSnapSetFailJsonKeysValidation(c *check.C) {
+func (s *snapSetSuite) TestSnapSetJSONStructureValidationUnhappy(c *check.C) {
```

Comment:
Since both this function and `TestSnapSetJSONValidationUnhappy` use the same `Parser` and `ParseArgs` (and there's not a `TestSnapSetJSONValidationHappy`), I think these should just be one function called `TestSnapSetJSONValidation`, which probably has two separate for-loops, the first one checks for invalid json, the second one checks for invalid keys (different errors from the same function calls).

Reply:
Alright! Done, thanks for the suggestion.